### PR TITLE
Issue: #336: Split "Administer comments"

### PR DESCRIPTION
**Pull request by docwilmot**
Fixes issue #336. Assumes persons with "administer comments" permission should also be able to "administer comment settings" but not the reverse.

### CODE REVIEW

File: core/modules/comment/comment.module [hunk 1]
```diff
@@ -379,6 +379,9 @@ function comment_permission() {
     'administer comments' => array(
       'title' => t('Administer comments and comment settings'),
     ),
```
Comment (quicksketch) [inline core/modules/comment/comment.module:381]: We need to change the title here to just "Administer comments" and remove the "and comment settings" part.

File: core/modules/comment/comment.module [hunk 2]
```diff
@@ -1205,7 +1208,7 @@ function comment_form_node_form_alter(&$form, $form_state) {
   $node = $form['#node'];
   $form['comment_settings'] = array(
     '#type' => 'fieldset',
-    '#access' => user_access('administer comments'),
+    '#access' => user_access('administer comment settings') || user_access('administer comments'),
```
Comment (quicksketch) [inline core/modules/comment/comment.module:1211]: I think the intention was to not allow users that are moderators ('administer comments') allow to modify the comment settings. So perhaps this should simply be just 'administer comment settings' and not either/or? Tests should need to be adjusted to work with this new permission requirement.

Comment (jenlampton) [inline core/modules/comment/comment.module:1211]: Yes that's right, only people with administer comment settings should have access to this form.

Comment (docwilmot) [inline core/modules/comment/comment.module:1211]: I imagined "administer comment settings" as the mere moderators, and "administer comments" as the admins. My way, admins get access to evrything comment-related including settings, moderators can only set settings. Your suggestion leaves the option for a person to have access to everything else comment-related but be unable to set settings, which seems counter-intuitive.
Or do you see 'administer comment settings' as the higher permission? is it that you want 'moderators' to only have access to the operations at '/admin/content/comment'? (view, edit, delete)

**Comment by quicksketch** [comment]
> I imagined "administer comment settings" as the mere moderators, and "administer comments" as the admins.

Yeah, it's the other way around, where comment moderators will have "administer comments" and only admins or site builders would have "administer comment settings". We're talking about "All settings provided by comment module" with this new permission, as individual comments don't have "settings".

It looks like this issue wasn't properly linked up to the original issue at https://github.com/backdrop/backdrop-issues/issues/336, so we ended up with a duplicate pull request at https://github.com/backdrop/backdrop/pull/478. :(

That PR also needs a little work however, as we need an upgrade path to split the single permission into two.

**Comment by docwilmot** [comment]
Sorry, will stop making mistakes eventually, I hope.
I understand your vision for the permissions now, but shouldnt that mean
that the PR should also change most of the code requiring "administer
comments" permission to require the "administer comment settings"
permission instead since that is the admin permission? Otherwise it doesnt
make sense.

**Comment by quicksketch** [comment]
So what should happen here is:
- Settings for enabling comments on node types should require "administer comment settings".
- Options for listing/publishing/unpublishing/deleting comments should require "administer comments".

Unless I'm missing something, the only place where we have any "comment settings" is on the node type form, so I think it's only necessary to change the setting in that place. Everywhere else, the "administer comments" permission is already the correct permission.

**Pull request closed without merging**